var/spack/repos/builtin/packages/braker/package.py

```diff
@@ -30,13 +30,14 @@ class Braker(Package):
        that combines the advantages of GeneMark-ET and AUGUSTUS"""
 
     homepage = "http://exon.gatech.edu/braker1.html"
-    url      = "http://bioinf.uni-greifswald.de/augustus/binaries/BRAKER1_v1.11.tar.gz"
+    url      = "http://bioinf.uni-greifswald.de/augustus/binaries/BRAKER2.tar.gz"
```

Comment:
Oh wait, will this actually work? If you run `spack info`, I think you'll see that it tries to download from:
http://bioinf.uni-greifswald.de/augustus/binaries/BRAKER2.1.0.tar.gz
http://bioinf.uni-greifswald.de/augustus/binaries/BRAKER1.11.tar.gz

Can you try the following:
```python
url      = "http://bioinf.uni-greifswald.de/augustus/binaries/BRAKER_v2.1.0.tar.gz"
...
def url_for_version(self, version):
    url = "http://bioinf.uni-greifswald.de/augustus/binaries/BRAKER{0}_v{1}.tar.gz"
    if version >= Version('2'):
        return url.format('', version)
    else:
        return url.format(version.up_to(1), version)
```
Then run `spack fetch` on both versions and make sure it correctly downloads them.

Reply:
The braker 1.11 is named as braker1_v(version) and it's at archive, 2.1.0 is named as braker_v(version) and will move to archive if new version releases. Do I need to set list_url for archive and a separete url for 1.11?

Reply:
If my `url_for_version` suggestion didn't work, you can try that.

Basically, we need to specify a tarball name for v1 and for v2. As for whether it is in archive or not, if you keep the default `url` without archive and keep the `list_url` with archive, it _should_ search both if it can't find it.

Reply:
Spack can fetch other version 2 in archive, but not version 1 with the url_for_version. Have no idea what's wrong. I'll try the separate url. This might be my last pull request in spack, Thank you for all your help!